it/lessons/basics/functions.md

```diff
@@ -90,6 +91,195 @@ iex> Length.of [1, 2, 3]
+
+Partiamo con una nuova funzione:
+
+```elixir
+defmodule Greeter2 do
+  def hello(%{name: person_name} = person) do
+    IO.puts "Hello, " <> person_name
+    IO.inspect person
+  end
+end
+```
+
+Ricorda che Elixir farà il _pattern match_ subito all'inizio.
+In questo case quindi, ogni lato dovrà "combaciare" contro il parametro in entrata e associarlo a qualsiasi dato che corrispondi.
+Diamo un'occhiata al lato destro prima:
+
+```elixir
+person = %{name: "Fred", age: "95", favorite_color: "Taupe"}
+```
+
+Ora, `person` è stato valutato ed assegnato alla mappa `fred`.
+Passiamo all prossimo `pattern-match`:
+
+```elixir
+%{name: person_name} = %{name: "Fred", age: "95", favorite_color: "Taupe"}
+```
+
+Questa funzione è uguale a quella originale nel module `Greeter1`, dove il `pattern match` viene applicato solamente sul nome.
+Quello che abbiamo ottenuto sono due variabili invece di una:
+
+1. `person`, si riferische a `%{name: "Fred", age: "95", favorite_color: "Taupe"}`
+2. `person_name`, si riferische a `"Fred"`
+
+Quindi ora, quando `Greeter2.hello/1` viene invocata, possiamo usare tutte le informazioni possibili su Fred.
+
+```elixir
+# chiamata con la mappa della persona intera
+...> Greeter2.hello(fred)
+"Hello, Fred"
+%{age: "95", favorite_color: "Taupe", name: "Fred"}
+# chiamata con una mappa contenente solo la chiave `:name`
+...> Greeter2.hello(%{name: "Fred"})
+"Hello, Fred"
+%{name: "Fred"}
+# chiamata con una mappa senza la chiave `:name`
+...> Greeter2.hello(%{age: "95", favorite_color: "Taupe"})
+** (FunctionClauseError) no function clause matching in Greeter2.hello/1
+
+    The following arguments were given to Greeter2.hello/1:
+
+        # 1
+        %{age: "95", favorite_color: "Taupe"}
+
+    iex:15: Greeter2.hello/1
+```
+
+Abbiamo visto che Elixir può eseguire il _pattern match_ a più profondità in quanto ogni parametro deve "combaciare" contro i dati in arrivo indipendentemente, lasciandoci con le variabili da chiamare separatamente nella nostra funzione.
+
+Se cambiamo l'ordine di `%{name: person_name}` e `person` nella lista, otterremo lo stesso risultato in quanto ognuno di essi corrisponde a `fred` per conto proprio.
+
+Scambiamo l'ordine della variabile e della mappa:
+
+```elixir
+defmodule Greeter3 do
+  def hello(person = %{name: person_name}) do
+    IO.puts "Hello, " <> person_name
+    IO.inspect person
+  end
+end
+```
+
+Ed invochiamo la funzione con lo stesso argomento usato in `Greeter2.hello/1`:
+
+```elixir
+...> Greeter3.hello(fred)
+"Hello, Fred"
+%{age: "95", favorite_color: "Taupe", name: "Fred"}
+```
+
+Ricorda che anche se sembra che `%{name: person_name} = person` stia eseguendo il _patter match_ tra `%{name: person_name}` e `person`, questo non è giusto! Tutt'e due i lati vengono usati per eseguire il pattern match contro il parametro in entrata.
+
+**Riepilogo** Le funzioni eseguiscono il pattern match dei parametri in entrata contro ogni argomento independentemente.
```

Comment:
* `eseguiscono` -> `eseguono`
* `independentemente` -> `indipendentemente`

Reply:
I knew `eseguiscono` didn't sound right! It was a very long translation and after a while I kinda lost my head I guess!
I will fix them as soon as I get a bit of free time! :D

Reply:
You did a huge/awesome work anyway ;-)

Reply:
fixed :)